src/mudlet.cpp

```diff
@@ -672,242 +376,613 @@ mudlet::mudlet()
-
-    // we historically use Alt on Windows and Linux, but that is uncomfortable on macOS
-#if defined(Q_OS_MACOS)
-    triggersKeySequence = QKeySequence(Qt::CTRL | Qt::Key_E);
-    showMapKeySequence = QKeySequence(Qt::CTRL | Qt::Key_M);
-    inputLineKeySequence = QKeySequence(Qt::CTRL | Qt::Key_L);
-    optionsKeySequence = QKeySequence(Qt::CTRL | Qt::Key_P);
-    notepadKeySequence = QKeySequence(Qt::CTRL | Qt::Key_N);
-    packagesKeySequence = QKeySequence(Qt::CTRL | Qt::Key_O);
-    modulesKeySequence = QKeySequence(Qt::CTRL | Qt::Key_I);
-    multiViewKeySequence = QKeySequence(Qt::CTRL | Qt::ALT | Qt::Key_V);
-    connectKeySequence = QKeySequence(Qt::CTRL | Qt::ALT | Qt::Key_C);
-    disconnectKeySequence = QKeySequence(Qt::CTRL | Qt::Key_D);
-    reconnectKeySequence = QKeySequence(Qt::CTRL | Qt::Key_R);
-#else
-    triggersKeySequence = QKeySequence(Qt::ALT | Qt::Key_E);
-    showMapKeySequence = QKeySequence(Qt::ALT | Qt::Key_M);
-    inputLineKeySequence = QKeySequence(Qt::ALT | Qt::Key_L);
-    optionsKeySequence = QKeySequence(Qt::ALT | Qt::Key_P);
-    notepadKeySequence = QKeySequence(Qt::ALT | Qt::Key_N);
-    packagesKeySequence = QKeySequence(Qt::ALT | Qt::Key_O);
-    modulesKeySequence = QKeySequence(Qt::ALT | Qt::Key_I);
-    multiViewKeySequence = QKeySequence(Qt::ALT | Qt::Key_V);
-    connectKeySequence = QKeySequence(Qt::ALT | Qt::Key_C);
-    disconnectKeySequence = QKeySequence(Qt::ALT | Qt::Key_D);
-    reconnectKeySequence = QKeySequence(Qt::ALT | Qt::Key_R);
-#endif
-    connect(this, &mudlet::signal_menuBarVisibilityChanged, this, &mudlet::slot_update_shortcuts);
-
-    mpSettings = getQSettings();
-    readLateSettings(*mpSettings);
-    // The previous line will set an option used in the slot method:
-    connect(mpMainToolBar, &QToolBar::visibilityChanged, this, &mudlet::slot_handleToolbarVisibilityChanged);
-
-#if defined(INCLUDE_UPDATER)
-    updater = new Updater(this, mpSettings);
-    connect(updater, &Updater::signal_updateAvailable, this, &mudlet::slot_updateAvailable);
-    connect(dactionUpdate, &QAction::triggered, this, &mudlet::slot_check_manual_update);
-#if defined(Q_OS_MACOS)
-    // ensure that 'Check for updates' is under the Applications menu per convention
-    dactionUpdate->setMenuRole(QAction::ApplicationSpecificRole);
-#else
-    connect(updater, &Updater::signal_updateInstalled, this, &mudlet::slot_update_installed);
-#endif // !Q_OS_MACOS
-#endif // INCLUDE_UPDATER
 
-    if (!mToolbarIconSize) {
-        setToolBarIconSize(mEnableFullScreenMode ? 2 : 3);
+    for (auto& translationFileName : qAsConst(translationFilesList)) {
+        QString languageCode(translationFileName);
+        languageCode.remove(QStringLiteral("mudlet_"), Qt::CaseInsensitive);
+        languageCode.remove(QStringLiteral(".qm"), Qt::CaseInsensitive);
+        int percentageTranslated = -1;
+
+        auto pMudletTranslator = new QTranslator();
```

Comment:
Use a `std::` smart pointer here instead that gets auto-deleted when it goes out of scope - modern C++ way and you don't have to worry about forgetting your `delete`'s.

Reply:
Is that anything like a `QPointer<T>` ? TBH I do not know about smart pointers in the STL...

It also points out to me that I have not arranged for the loaded `QTranslator`s to be `delete`d when the application exits - though depending on exactly where in the `mudlet` class we are here perhaps that is the point you are trying to make.... :worried:

Reply:
No it's like `QScopedPointer` but worse. Better to use `std::unique_ptr`.

Reply:
:thought_balloon:  Humm, interesting.... :wink:

Reply:
Ah, you still have to create a normal (`auto` ?) pointer but then you pass that to the `std::unique_ptr` constructor:
```
        auto* pTranslator = new QTranslator();
        std::unique_ptr<QTranslator> pQtTranslator(pTranslator);

```
however the latter will take care of the destruction.

Reply:
See `std::make_unique`: https://agateau.com/2019/using-std-unique-ptr-with-qt/

Then you can do it in one line!